src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -70,11 +70,15 @@ SubmissionType getSubmissionType(String requestId, String overseasEntityId) thro
 
         String entityNumber = submissionOpt.get().getEntityNumber();
         if (submissionOpt.get().isForUpdate()) {
-            ApiLogger.infoContext(requestId, String.format("Submission with overseas entity number %s found",
+            ApiLogger.infoContext(requestId, String.format("Update submission with overseas entity number %s found",
                     entityNumber));
             return SubmissionType.UPDATE;
+        } else if (submissionOpt.get().isForRemove()) {
+            ApiLogger.infoContext(requestId, String.format("Remove submission with overseas entity number %s found",
+                    entityNumber));
+            return SubmissionType.REMOVE;
         } else {
-            ApiLogger.infoContext(requestId, "Submission without overseas entity number found");
+            ApiLogger.infoContext(requestId, "Registration submission without overseas entity number found");
```

Comment:
The Registration info message might be misleading, it makes it sound like some Registration submission can have an entity number.

Reply:
Yes, agree. It should probably just be "Registration submission found".